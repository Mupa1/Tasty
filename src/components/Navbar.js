import React from 'react';
import { Link } from 'react-router-dom';

const Navbar = () => (
  <header className="bg-primary">
    <nav className="text-center">
      <Link to="/">
        <h1 className="text-white">Tasty Recipes</h1>
      </Link>
    </nav>
  </header>
);

export default Navbar;
