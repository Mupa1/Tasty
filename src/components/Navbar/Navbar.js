import React from 'react';
import { Link, BrowserRouter } from 'react-router-dom';

import './Navbar.module.css';

const Navbar = () => (
  <header className="bg-success text-center">
    <nav data-testid="navbar">
      <BrowserRouter>
        <Link to="/">
          <h1 className="text-white">Tasty Recipes</h1>
        </Link>
      </BrowserRouter>
    </nav>
  </header>
);

export default Navbar;
