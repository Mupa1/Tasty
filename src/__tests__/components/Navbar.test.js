import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render, screen, cleanup } from '@testing-library/react';

import Navbar from '../../components/Navbar/Navbar';

afterEach(cleanup);

it('matches Navbar snapshot', () => {
  const tree = renderer
    .create(
      <Router>
        <Navbar />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains Tasty Recipes logo', () => {
  render(
    <Router>
      <Navbar />
    </Router>,
  );
  expect(screen.getByText('Tasty Recipes')).toBeInTheDocument();
});

it('should not render a different logo', () => {
  const { getAllByTestId } = render(
    <Router>
      <Navbar />
    </Router>,
  );
  const result = getAllByTestId('navbar');
  expect(result[0]).not.toHaveTextContent('nothing');
});
