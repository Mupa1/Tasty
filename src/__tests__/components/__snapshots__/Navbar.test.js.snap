// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Navbar snapshot 1`] = `
<header
  className="bg-success text-center"
>
  <nav
    data-testid="navbar"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <h1
        className="text-white"
      >
        Tasty Recipes
      </h1>
    </a>
  </nav>
</header>
`;
